homeassistant/components/sensor/zha.py

```diff
@@ -111,3 +132,155 @@ def state(self):
             return 'unknown'
 
         return round(float(self._state) / 100, 1)
+
+
+class GenericBatterySensor(Sensor):
+    """ZHA generic battery sensor."""
+
+    value_attribute = 32
+    battery_sizes = {
+        0: 'No battery',
+        1: 'Built in',
+        2: 'Other',
+        3: 'AA',
+        4: 'AAA',
+        5: 'C',
+        6: 'D',
+        7: 'CR2',
+        8: 'CR123A',
+        255: 'Unknown'
+    }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return '%'
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+        _LOGGER.debug("%s async_update", self.entity_id)
+
+        result = yield from zha.safe_read(
+            self._endpoint.power,
+            ['battery_size', 'battery_quantity', 'battery_voltage']
+        )
+        self._device_state_attributes['battery_size'] = \
+            self.battery_sizes.get(
+                result.get('battery_size', 255),
+                'Unknown'
+            )
+        self._device_state_attributes['battery_quantity'] = result.get(
+            'battery_quantity', 'Unknown')
+        self._state = result.get('battery_voltage', self._state)
+
+
+class CentraliteBatterySensor(GenericBatterySensor):
+    """ZHA battery sensor."""
+
+    # currently restricted to centralite sensors because the value
+    # conversion is specific to centralite sensors.
+
+    minVolts = 15
+    maxVolts = 28
+    values = {
+        28: 100,
+        27: 100,
+        26: 100,
+        25: 90,
+        24: 90,
+        23: 70,
+        22: 70,
+        21: 50,
+        20: 50,
+        19: 30,
+        18: 30,
+        17: 15,
+        16: 1,
+        15: 0
+    }
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        if self._state == 'unknown':
+            return 'unknown'
```

Comment:
Return `None` to represent unknown state.

Reply:
Why are multiple input values mapped to the same output percentage? 
28, 27 and 26 are 100%
Why not 28=100, 27=97, 26=93? (or something like that)

Reply:
@ryanwinter please move your comment to a general PR comment. I don't see that it's related to my comment or this code line.